Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -29,77 +31,107 @@ namespace SonarScanner.MSBuild.PreProcessor.Test
+            GetHeader(PreprocessorObjectFactory.CreateHttpClient("da39a3ee5e6b4b0d3255bfef95601890afd80709", string.Empty, null, null), HttpRequestHeader.Authorization)
+                .Should().Be("Basic ZGEzOWEzZWU1ZTZiNGIwZDMyNTViZmVmOTU2MDE4OTBhZmQ4MDcwOTo=");
+            GetHeader(PreprocessorObjectFactory.CreateHttpClient("admin", "password", null, null), HttpRequestHeader.Authorization).Should().Be("Basic YWRtaW46cGFzc3dvcmQ=");
+        }
 
-        #region Private methods
+        [TestMethod]
+        public void CreateHttpClient_UserAgent()
+        {
+            var userAgent = GetHeader(PreprocessorObjectFactory.CreateHttpClient(null, null, null, null), HttpRequestHeader.UserAgent);
+
+            var scannerVersion = typeof(WebClientDownloaderTest).Assembly.GetName().Version.ToDisplayString();
+            userAgent.Should().Be($"ScannerMSBuild/{scannerVersion}");
+        }
 
-        private ProcessedArgs CreateValidArguments()
+        [TestMethod]
+        public void CreateHttpClient_SemicolonInUsername()
         {
-            var logger = new TestLogger();
-            var cmdLineArgs = new Common.ListPropertiesProvider();
-            cmdLineArgs.AddProperty(Common.SonarProperties.HostUrl, "http://foo");
-
-            var validArgs = new ProcessedArgs("key", "name", "verions", "organization", false,
-                cmdLineArgs,
-                new Common.ListPropertiesProvider(),
-                EmptyPropertyProvider.Instance,
-                logger);
-            return validArgs;
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient("user:name", null, null, null);
+            act.Should().ThrowExactly<ArgumentException>().WithMessage("username cannot contain the ':' character due to basic authentication limitations");
+        }
+
+        [TestMethod]
+        public void CreateHttpClient_AccentsInUsername()
+        {
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient("héhé", "password", null, null);
+            act.Should().ThrowExactly<ArgumentException>().WithMessage("username and password should contain only ASCII characters due to basic authentication limitations");
+        }
+
+        [TestMethod]
+        public void CreateHttpClient_AccentsInPassword()
+        {
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient("username", "héhé", null, null);
+            act.Should().ThrowExactly<ArgumentException>().WithMessage("username and password should contain only ASCII characters due to basic authentication limitations");
         }
 
-        #endregion Private methods
+        [TestMethod]
+        public void CreateHttpClient_UsingClientCert()
+        {
+            Action act = () => PreprocessorObjectFactory.CreateHttpClient(null, null, "certtestsonar.pem", "dummypw");
+            act.Should().NotThrow();
+        }
```

Comment:
Let's add same UT with `"nonexistentcert.pem"` and assert that it will throw the expected exception type